bench/db_bench.ml

```diff
@@ -323,51 +339,60 @@ let init () =
   Log.app (fun l -> l "Log size: %d." log_size);
   populate ()
 
-let run input =
+let run input json_file_name =
   init ();
   Log.app (fun l -> l "\n");
   Log.app (fun l -> l "Fill in random order");
-  let rw = Index.write_random () in
+  let rw, json_element = Index.write_random "fill_random" in
+  let output_json = [ json_element ] in
   let bench_list =
     [
-      ( (fun () -> Index.read_random rw),
+      ( (fun () -> Index.read_random rw "read_random"),
         [ `Read `RW; `All; `Minimal; `Index ],
         "RW Read in random order" );
-      ( (fun () -> Index.ro_read_random rw),
+      ( (fun () -> Index.ro_read_random rw "read_only_random"),
         [ `Read `Ro; `All; `Minimal; `Index ],
         "RO Read in random order" );
-      ( (fun () -> Index.read_absent rw),
+      ( (fun () -> Index.read_absent rw "read_absent"),
         [ `Read `Absent; `All; `Minimal; `Index ],
         "Read 1000 absent values" );
-      ( (fun () -> Index.read_seq rw),
+      ( (fun () -> Index.read_seq rw "read_seq"),
         [ `Read `Seq; `All; `Index ],
         "Read in sequential order (increasing order of hashes for index" );
-      ( Index.write_seq,
+      ( (fun () -> Index.write_seq "fill_seq"),
         [ `Write `IncKey; `All; `Index ],
         "Fill in increasing order of keys" );
-      ( Index.write_seq_hash,
+      ( (fun () -> Index.write_seq_hash "fill_seq_hash"),
         [ `Write `IncHash; `All; `Index ],
         "Fill in increasing order of hashes" );
-      ( Index.write_rev_seq_hash,
+      ( (fun () -> Index.write_rev_seq_hash "fill_rev_seq_hash"),
         [ `Write `DecHash; `All; `Index ],
         "Fill in decreasing order of hashes" );
-      ( Index.write_sync,
+      ( (fun () -> Index.write_sync "fill_sync"),
         [ `Write `Sync; `All; `Index ],
         "Fill in random order and sync after each write" );
-      ((fun () -> Index.overwrite rw), [ `OverWrite; `All; `Index ], "OverWrite");
-      ((fun () -> lmdb_benchmarks ()), [ `Lmdb ], "Run lmdb benchmarks");
+      ( (fun () -> Index.overwrite rw "overwrite"),
+        [ `OverWrite; `All; `Index ],
+        "OverWrite" );
+      ((fun () -> lmdb_benchmarks ()), [ `Lmdb; `All ], "Run lmdb benchmarks");
     ]
   in
   let match_input ~bench ~triggers ~message =
     if List.mem input triggers then
       let () = Log.app (fun l -> l "\n %s" message) in
       bench ()
-    else ()
+    else []
+  in
+  let output_json =
+    List.fold_left
+      (fun foo (bench, triggers, message) ->
```

Comment:
personal preferences here, but rename `foo` in `acc` or `output_json` or something suggestive